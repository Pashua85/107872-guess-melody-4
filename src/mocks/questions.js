const AVATAR_URL = `https://api.adorable.io/avatars/17`;

export default [
  {
    type: `artist`,
    questionText: `Кто исполняет эту песню?`,
    song: {
      artist: `Gregory Porter`,
      src: `audio/gregory_porter_ hey_laura.mp3`
    },
    answers: [{
      picture: `${AVATAR_URL}/${Math.random()}`,
      artist: `Kurt Elling`
    }, {
      picture: `${AVATAR_URL}/${Math.random()}`,
      artist: `Gregory Porter`
    }, {
      picture: `${AVATAR_URL}/${Math.random()}`,
      artist: `Tony Bennett`
    }]
  },
  {
    type: `genre`,
    questionText: `Выберите музыку барокко`,
    genre: `baroque`,
    answers: [{
      src: `audio/handel_concerto_grosso_in_g_minor_allegro.mp3`,
      genre: `baroque`
    }, {
      src: `audio/haydn_symphony_no_6_finale.mp3`,
      genre: `classical`
    }, {
      src: `audio/franz_liszt_hungarian_rhapsody_no_11.mp3`,
      genre: `romantic`
    }, {
      src: `audio/philip_glass_string_quartet_no_2_company_movement_2`,
      genre: `minimalism`
    }]
  }
];
