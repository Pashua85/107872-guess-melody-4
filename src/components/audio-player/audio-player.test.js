import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AudioPlayer from './audio-player';
import questions from '../../mocks/test-questions';

Enzyme.configure({
  adapter: new Adapter()
});

describe(`AudioPlayer`, () => {
  it(`should render correctly with one track`, () => {
    const wrapper = shallow(
        <AudioPlayer
          tracks={questions[0].tracks}
          type={questions[0].type}
        />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it(`should render correctly with 4 tracks`, () => {
    const wrapper = shallow(
        <AudioPlayer
          tracks={questions[1].tracks}
          type={questions[1].type}
          onInputChange={() => {}}
        />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
