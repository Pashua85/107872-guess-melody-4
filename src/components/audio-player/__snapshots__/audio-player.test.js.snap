// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer should render correctly with 4 tracks 1`] = `ShallowWrapper {}`;

exports[`AudioPlayer should render correctly with one track 1`] = `ShallowWrapper {}`;
